support/cdi/src/main/java/org/apache/shiro/cdi/literal/AnyLiteral.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.shiro.cdi.literal;
+
+import javax.enterprise.inject.Any;
+import javax.enterprise.util.AnnotationLiteral;
+
```

Comment:
This going to be used by cdi users and any cdi user who don't know what is "Any" in cdi should not learn from shiro ! Only doc can be helpful but not essential would be "traces" for cdi extension developers if this doc really going to speak to shiro users then only cdi reference links can be given other than that will be redundant. This pull request does not change the behaviour of shiro it only makes more pluggable to any other component. For doc sake but not for this AnyLiteral but for this extension request what can be done is more cdi based usage tests for authentication, authorization, filtering and such ... Those test can be very useful for extension user starters and also for cdi users which don't know much about shiro. They can quickly play with shiro.